x-pack/plugins/security_solution/public/management/cypress/e2e/artifact_tabs_in_policy_details.cy.ts

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getEndpointSecurityPolicyManager } from '../../../../scripts/endpoint/common/roles_users/endpoint_security_policy_manager';
+import { getArtifactsListTestsData } from '../fixtures/artifacts_page';
+import { removeAllArtifacts } from '../tasks/artifacts';
+import { loadEndpointDataForEventFiltersIfNeeded } from '../tasks/load_endpoint_data';
+import { login, loginWithCustomRole, loginWithRole, ROLE } from '../tasks/login';
+import { performUserActions } from '../tasks/perform_user_actions';
+
+const loginWithPrivilegeAll = () => {
+  loginWithRole(ROLE.endpoint_security_policy_manager);
+};
+
+const loginWithPrivilegeRead = (privilegePrefix: string) => {
+  const roleWithArtifactReadPrivilege = getRoleWithArtifactReadPrivilege(privilegePrefix);
+  loginWithCustomRole('roleWithArtifactReadPrivilege', roleWithArtifactReadPrivilege);
+};
+
+const loginWithPrivilegeNone = (privilegePrefix: string) => {
+  const roleWithoutArtifactPrivilege = getRoleWithoutArtifactPrivilege(privilegePrefix);
+
+  loginWithCustomRole('roleWithoutArtifactPrivilege', roleWithoutArtifactPrivilege);
+};
```

Comment:
This actually has `read` privileges for the artifact even though it is supposed to be `none`? 🤔  I would suggest you use the `t1_analyst` for `none` privilege checks if you need it.

Reply:
this should have `none`, it calls the `getRoleWithoutArtifactPrivilege()` function that removes the `all` privilege. it uses `endpointSecurityPolicyManagerRole` as a base, and that has only `all` privileges, so no `read` privileges remains. or do I miss something?

Reply:
Also, I did a console log on `roleWithoutArtifactPrivilege` and it shows all the privilges even though it should have filtered it out. Also I tested with loging in as `endpoint_security_policy_manager` within `loginWithPrivilegeNone`, the tests pass in that case. In conclusion, the `None` test is not testing correctly.

Reply:
You are right, thanks for looking into it so thoroughly! I forgot to wait for everything to be loaded. Here's the fix, it wait for the tabs to show up instead of just waiting for the container: 0725a92
(This fails with the security policy manager role, but passes with the `none` role)

Regarding `roleWithoutArtifactPrivilege`: also printed the results to the console, and it looks good on my end: the one privilege that should be filtered out is missing.

Reply:
Regarding creating a new role vs removing privileges one by one: I agree that creating a new role is cleaner, but I deliberately chose to programmatically disable one privilege at a time: to make sure that they are independent, and if you remove only one, it still disables the feature. It is not perfect, still does not test every combination of the privileges, but still more than having all-`none`, all-`read` and all-`all` roles. What do you think of this?

Reply:
I see what you mean. I was testing with the assumption that I don't have any artifact privilege so I should just have `minimal_all` and `policy_management_read`. I think it is fine as you've now, where you don't have the `read` privilege for that specific artifact you're testing. 👍